src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -178,6 +180,7 @@ func preparePVCResource(
 					corev1.ResourceStorage: resource.MustParse(defaultDatadirCapacity),
 				},
 			},
+			VolumeMode: &fileSystemMode,
```

Comment:
this is not nice, I'll handle this better in follow up PR. This is default that's set by apiserver, but if we don't provide it, the patchmaker thinks we need to patch the object.

I'll handle this better next week :)